minio/minio
# Fix `updateGroupMembershipsForLDAP` behavior with unicode

## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
`updateGroupMembershipsForLDAP` now queries LDAP server with actual DNs rather than internal normalized forms.

## Motivation and Context
Fixes #20112 

## How to test this PR?
Create access key for LDAP user with non-ASCII characters that gets its policy from a group. Let run until `updateGroupMembershipsForLDAP` runs (~40 minutes, 60 minutes max). Previously, group memberships would be deleted and access key would no longer work. Now, group memberships remain intact.

## Types of changes

## Checklist:
